framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -754,17 +755,48 @@ public AnnotationFormatter getAnnotationFormatter() {
      *
      * <p>
      * Subclasses cannot override this method; they should override
-     * {@link #createSupportedTypeQualifiers createSupportedTypeQualifiers} instead.
-
+     * {@link #createSupportedTypeQualifiers createSupportedTypeQualifiers}
+     * instead.
+     *
      * @see #createSupportedTypeQualifiers()
      *
-     * @return the type qualifiers supported this processor, or an empty
-     * set if none
+     * @return an immutable HashSet of the supported type qualifiers, or an
+     *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
         return supportedQuals;
     }
 
+    /**
+     * Defines alphabetical sort ordering for qualifiers
+     */
+    private static final Comparator<Class<? extends Annotation>> QUALIFIER_SORT_ORDERING
+    = new Comparator<Class<? extends Annotation>>() {
+        @Override
+        public int compare(Class<? extends Annotation> a1, Class<? extends Annotation> a2) {
+            return a1.getCanonicalName().compareTo(a2.getCanonicalName());
+        }
+    };
+
+    /**
+     * Returns an alphabetically sorted immutable list of the type qualifiers
+     * supported by this checker. This method is useful for debug printing
+     * purposes, but otherwise returns the same qualifiers as
+     * {@link #getSupportedTypeQualifiers()}.
+     *
+     * @see #getSupportedTypeQualifiers()
+     *
+     * @return an immutable sorted list of the supported type qualifiers, or an
+     *         immutable empty list if no qualifiers are supported
+     */
+    public final List<Class<? extends Annotation>> getSortedSupportedTypeQualifiers() {
```

Comment:
I don't see the value of having this public method in the ATF. I would fold the logic into getSortedQualifierNames or have a private method there.